packages/aws-cdk/lib/api/logs/register-cloudwatch-logs.ts

```diff
@@ -0,0 +1,90 @@
+import * as cxapi from '@aws-cdk/cx-api';
+import { Mode, SdkProvider, ISDK } from '../aws-auth';
+import { EvaluateCloudFormationTemplate, LazyListStackResources } from '../evaluate-cloudformation-template';
+
+// resource types that have associated CloudWatch Log Groups that should
+// _not_ be monitored
+const IGNORE_LOGS_RESOURCE_TYPES = ['AWS::EC2::FlowLog', 'AWS::CloudTrail::Trail', 'AWS::CodeBuild::Project'];
+
+// resource types that will create a cloudwatch log group
+// with a specific name if one is not provided
+const RESOURCE_TYPES_WITH_IMPLICIT_LOGS = ['AWS::Lambda::Function'];
+
+export async function registerCloudWatchLogGroups(
+  sdkProvider: SdkProvider,
+  stackArtifact: cxapi.CloudFormationStackArtifact,
+): Promise<{ env: cxapi.Environment, sdk: ISDK, logGroupNames: string[] }> {
+
+
+  const resolvedEnv = await sdkProvider.resolveEnvironment(stackArtifact.environment);
+  const sdk = await sdkProvider.forEnvironment(resolvedEnv, Mode.ForReading);
+
+  const listStackResources = new LazyListStackResources(sdk, stackArtifact.stackName);
+  const evaluateCfnTemplate = new EvaluateCloudFormationTemplate({
+    stackArtifact,
+    parameters: {},
+    account: resolvedEnv.account,
+    region: resolvedEnv.region,
+    partition: (await sdk.currentAccount()).partition,
+    urlSuffix: sdk.getEndpointSuffix,
+    listStackResources,
+  });
+
+  const resources = await listStackResources.listStackResources();
+
+  const logGroupLogicalIds = new Set<string>();
+  // map of logicalId to AWS service name
+  // e.g. mylambdaFunction: lambda
+  const logicalIdsOfImplicitLogServices = new Map<string, string>();
+  //
+  // do a first pass at identifying all log groups
+  for (const resource of resources) {
```

Comment:
Hmm. Why are we actually iterating through the list of resources from the service? We have the generated CDK template with the `stackArtifact.template` parameter, we should probably iterate through that instead.

Reply:
Good call. Updated.